megaparse/checker/markdown_processor.py

```diff
@@ -0,0 +1,211 @@
+#         """
+
+#         def should_remove(paragraph):
+#             if self.remove_pagination and "Page" in paragraph:
+#                 return True
+#             return any(word in paragraph for word in header_components_count.keys())
+
+#         return [paragraph for paragraph in paragraphs if not should_remove(paragraph)]
+
+#     def merge_tables(self, md_content: str) -> str:
+#         """
+#         Merge tables inside Markdown content.
+
+#         Args:
+#             md_content (str): Markdown content
+
+#         Returns:
+#             str: Merged tables
+#         """
+#         md_content = md_content.replace("|\n\n|", "|\n|")
+#         return md_content
+
+#     def save_cleaned_result(self, cleaned_result: str, output_path: str) -> None:
+#         """
+#         Save the cleaned paragraphs to a markdown file.
+
+#         Args:
+#             cleaned_result (str): Cleaned result
+#             output_path (str): Output path
+#         """
+#         with open(output_path, "w") as f:
+#             f.write(cleaned_result)
+
+#     def remove_header_llm(self):
+#         llm = ChatOpenAI(model="gpt-4o", api_key=os.getenv("OPENAI_API_KEY"))
+#         # Define the prompt
+#         messages = [
+#             (
+#                 "system",
+#                 "You are a document cleaner and you are used to remove repetitive headers / footer from parsed files in markdown.",
+#             ),
+#         ]
+
+#         prompt = f"""You are a document cleaner and you are used to remove repetitive headers / footer from parsed files in markdown.
+#         Here is a md file : "{self.md_result}"
+#         I want you to identify repetitive texts that could be associate to a document header and footer. Please identify the headers, the footer and remove them from the document.
+#         Answer with only the cleaned document in markdown format.
+#         Result : """
+
+#         messages.append(("human", self.md_result))  # type: ignore
+
+#         result = llm.invoke(messages)
+
+#         return result.content
+
+#     def process(self, gpt4o_cleaner=False) -> str:
+#         """
+#         Process the markdown result by removing duplicate paragraphs and headers.
+
+#         Args:
+#             gpt4o_cleaner (bool, optional): GPT-4o cleaner. Defaults to False.
+
+#         Returns:
+#             str: Cleaned result
+#         """
+#         if gpt4o_cleaner:
+#             cleaned_result = self.remove_header_llm()
+
+#         else:
+#             pages = self.split_into_pages()
+#             paragraphs = self.split_into_paragraphs(pages)
+#             # other_pages_paragraphs = self.split_into_paragraphs(pages[1:])
+
+#             cleaned_paragraphs, duplicate_paragraphs = self.remove_duplicates(
+#                 paragraphs
+#             )
+#             header_components_count = self.identify_header_components(
+#                 duplicate_paragraphs
+#             )
+
+#             if self.strict:
```

Comment:
Remove the deadcode

Reply:
It's not dead code it is something to keep for when i tackle FormatCheckers